plugins/inputs/signalfx_metadata/process-info.go

```diff
@@ -48,159 +62,178 @@ func (s *ProcessInfo) GetTop() ([]byte, error) {
 				}
 			}
 		}
+
+		// use separate go routine to push processes on to worker threads
 		for pid, proc := range s.processes {
 			// Remove dead processes from process list
 			if _, in := pidList[pid]; !in {
 				delete(s.processes, pid)
 			} else {
-				pid64 := int64(pid)
-				stringPid := strconv.FormatInt(pid64, 10)
-				top[stringPid] = GetProcessInfo(proc)
+				s.processIn <- &workerInProcess{
+					pid:  pid,
+					proc: proc,
+					f:    s.GetProcessInfo,
+					out:  output,
+				}
 			}
 		}
-		if js, er := json.Marshal(top); er == nil {
-			compressed := compressByteArray(js)
-			base64ed := base64.StdEncoding.EncodeToString(compressed)
-			response["t"] = base64ed
 
+		// wait for all processes to return
+		count := 0
+		for msg := range output {
+			top[strconv.FormatInt(int64(msg.pid), 10)] = msg.out
+			count++
+			if count == len(pids) {
```

Comment:
This seems unnecessary.  Once the `output` var is lost the channel will be automatically closed and deallocated.

Reply:
I could just break out of the for loop, or I could use the channel close to control flow.  Since the for loop is ranging over the channel it seems appropriate to me.